### PR TITLE
Use column name alias for projection expressions in the DynamoDB adapter

### PR DESCRIPTION
**Pull request by brfrn169**
This is for the issue reported in the following:
https://stackoverflow.com/questions/68587760/syntax-error-when-using-scalardb-with-dynamodb-reserved-words-as-column-names

To fix the issue, I think we need to use a column name alias for the projection expressions.

Please take a look!

### CODE REVIEW

File: core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java [hunk 1]
```diff
@@ -163,6 +164,27 @@ public SelectStatementHandler(DynamoDbClient client, DynamoTableMetadataManager
...
+    for (int i = 0; i < selection.getProjections().size(); i++) {
+      String alias = DynamoOperation.COLUMN_NAME_ALIAS + i;
+      projections.add(alias);
+      columnMap.put(alias, selection.getProjections().get(i));
+    }
+    String projectionExpression = String.join(",", projections);
+
+    if (builder instanceof GetItemRequest.Builder) {
+      ((GetItemRequest.Builder) builder).projectionExpression(projectionExpression);
+      ((GetItemRequest.Builder) builder).expressionAttributeNames(columnMap);
```
[the remark this answers is not in the document]
Comment (feeblefakie) [reply core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java:181]: Thank you!
It looks like `builder.projectionExpression()` was already properly called in the original code but `builder.expressionAttributeNames()` for the columnMap was missing so I understand that this PR adds `builder.expressionAttributeNames()` with the columnMap.  Am I correct?

**Review by feeblefakie — APPROVED** [review]

File: core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (brfrn169) [reply core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java:181]: We passed the original names to `builder.projectionExpression()` in the original code, but we pass the aliases in the new code. That's the difference regarding `builder.projectionExpression()`. Also, we pass the columnMap to `builder.expressionAttributeNames()` in the new code.

**Review by yito88 — APPROVED** [review]
LGTM! thanks!

File: core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (feeblefakie) [reply core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java:181]: Ah, got it. Thank you!

**Review by feeblefakie — APPROVED** [review]

**Pull request merged**